cruddy/__init__.py

```diff
@@ -52,67 +83,135 @@ def _replace_decimals(self, obj):
-        id = item.get('id')
-        if id is None:
+    def _check_supported_op(self, op_name, response):
+        if op_name not in self.supported_ops:
             response['status'] = 'error'
-            response['message'] = 'delete requires an id'
-        else:
-            response['data'] = self._table.delete_item(Key={'id': id})
+            response['error_type'] = 'UnsupportedOperation'
+            response['message'] = 'unsupported operation: {}'.format(op_name)
+            return False
+        return True
 
-    def _get(self, item, response):
-        id = item.get('id')
-        if id is None:
+    def _call_ddb_method(self, method, kwargs, response):
+        try:
+            response['raw_response'] = method(**kwargs)
+        except Exception as e:
             response['status'] = 'error'
-            response['message'] = 'get requires an id'
-        else:
-            item = self._table.get_item(Key={'id': id})['Item']
-            response['data'] = self._replace_decimals(item)
+            response['error_type'] = e.__class__.__name__
+            response['error_message'] = str(e)
+
+    def _new_response(self):
+        return dict(status='success', data=None)
+
+    def _prepare_response(self, response):
+        if response['status'] == 'success':
+            if 'raw_response' in response:
+                md = response['raw_response']['ResponseMetadata']
+                response['response_metadata'] = md
+                del response['raw_response']
+        return response
+
+    def list(self):
+        response = self._new_response()
+        if self._check_supported_op('list', response):
+            self._call_ddb_method(self.table.scan, {}, response)
+            if response['status'] == 'success':
+                response['data'] = self._replace_decimals(
+                    response['raw_response']['Items'])
+        return self._prepare_response(response)
+
+    def get(self, id, decrypt=False):
+        response = self._new_response()
+        if self._check_supported_op('list', response):
+            if id is None:
+                response['status'] = 'error'
+                response['error_type'] = 'IDRequired'
+                response['message'] = 'get requires an id'
+            else:
+                params = {'Key': {'id': id},
+                          'ConsistentRead': True}
+                self._call_ddb_method(self.table.get_item, params, response)
+                if response['status'] == 'success':
+                    item = response['raw_response']['Item']
+                    if decrypt:
+                        self._decrypt(item)
+                    response['data'] = self._replace_decimals(item)
+        return self._prepare_response(response)
+
+    def create(self, item):
+        response = self._new_response()
+        if self._check_supported_op('create', response):
+            item['id'] = str(uuid.uuid4())
+            item['created_at'] = int(time.time() * 1000)
```

Comment:
`int(time.time() * 1000)` should probably be a method somewhere since it is repeated and is our standard.